src/core_plugins/elasticsearch/index.js

```diff
@@ -15,8 +16,6 @@ const DEFAULT_REQUEST_HEADERS = [ 'authorization' ];
 
 module.exports = function ({ Plugin }) {
   return new Plugin({
-    require: ['kibana'],
```

Comment:
The elasticsearch plugin needs kibana to get configuration values for `kibana.index` and friends, kibana needs the elasticsearch plugin now to register mappings.  We can't have circular dependencies however.  [Configuration is registered before plugin initialization now](https://github.com/elastic/kibana/blob/master/src/server/kbn_server.js#L64), so this case works out, but it's not great.  [This](https://github.com/elastic/kibana/blob/eef8bb58d6da1aef4412dd68ccf6551d32d63545/src/plugins/elasticsearch/index.js#L7) is the original commit.

@spalger any extra thoughts/do you have concerns with this?